src/get-student-id.js

```diff
@@ -1,13 +1,12 @@
 import uuid from "uuid/v4";
 
-export const getStudentId = app =>
-  new Promise(resolve => {
-    app.userData.read("studentId", data => {
-      if (data && data.studentId) {
-        resolve(data.studentId);
-      } else {
-        data = { studentId: uuid() };
-        app.userData.write("studentId", data, () => resolve(data.studentId));
-      }
-    });
-  });
+const createNewId = dataStore => {
+  const id = uuid();
+  return dataStore.write("studentId", id).then(() => id, () => id);
```

Comment:
Is there a difference between `then(() => id, () => id)` and `then(() => id).catch(() => id)`?

Reply:
They are different but not in a way that would matter here since we're not doing an operation that can throw. This StackOverflow answer has an explanation of the difference: https://stackoverflow.com/a/33278420